### PR TITLE
 SI-6591 Reify and path-dependent types 

### PR DESCRIPTION
**Pull request by xeno-by**
review @retronym @paulp

### CODE REVIEW

**Comment by 4e6** [comment]
@xeno-by, thank you very much for your help! Also, I am glad you fixed my dirty hack with `PredefObject`

**Comment by JamesIry** [comment]
ping @paulp for review

File: src/reflect/scala/reflect/internal/Flags.scala [hunk 1]
```diff
@@ -274,7 +274,7 @@ class Flags extends ModifierFlags {
    *  from Modifiers.  Others which may be applied at creation time are:
    *  SYNTHETIC.
    */
-  final val ValueParameterFlags = BYNAMEPARAM | IMPLICIT | DEFAULTPARAM
+  final val ValueParameterFlags = BYNAMEPARAM | IMPLICIT | DEFAULTPARAM | STABLE
```
Comment (paulp) [inline src/reflect/scala/reflect/internal/Flags.scala:277]: Although this appears correct, I would be surprised if it did not have unforeseen effects.

**Comment by paulp** [comment]
LGTM, but as mentioned, I have a feeling that some future thing will lead back to the additional of STABLE there.

**Comment by xeno-by** [comment]
PLS REBUILD/pr-checkin-per-commit@ce867c7

**Pull request merged**